pkg/sqlite/query.go

```diff
@@ -571,75 +613,153 @@ func (s *SQLQuerier) GetImagesForBundle(ctx context.Context, csvName string) ([]
 }
 
 func (s *SQLQuerier) GetApisForEntry(ctx context.Context, entryID int64) (provided []*api.GroupVersionKind, required []*api.GroupVersionKind, err error) {
-	providedQuery := `SELECT DISTINCT api.group_name, api.version, api.kind, api.plural FROM api
-		 	  		  INNER JOIN channel_entry ON channel_entry.operatorbundle_name = api_provider.operatorbundle_name
-					  INNER JOIN operatorbundle ON operatorbundle.name=channel_entry.operatorbundle_name
-		 	  		  INNER JOIN api_provider ON (api.group_name=api_provider.group_name AND api.version=api_provider.version AND api.kind=api_provider.kind AND operatorbundle.name=api_provider.operatorbundle_name)
-			  		  WHERE channel_entry.entry_id=?`
+	groups := map[string]struct{}{}
+	kinds := map[string]struct{}{}
+	versions := map[string]struct{}{}
+
+	providedQuery := `SELECT DISTINCT properties.value FROM properties 
```

Comment:
Would it be simpler not to collapse duplicate properties at this layer (or maybe at all)? My preference would be to keep properties as opaque as possible at the storage layer especially, and there's something unsatisfying about getting a different number of properties out on the read end than there were on the write end.

Besides, for a property type with JSON-encoded values, two strings could represent different but equivalent encodings of a single object.

Reply:
I'm not concerned about your second point because `json.Marshal` will sort the map keys. We can guarantee the JSON will be the same if the map has the same items.